### PR TITLE
Bug 2196912: Add missing and fix confusing "CPU | Memory" help texts

### PR DESCRIPTION
**Pull request by hstastna**
## 📝 Description

**Fixes:**
https://bugzilla.redhat.com/show_bug.cgi?id=2196912

Add missing "CPU | Memory" helper texts in the following pages:
- _Catalog_ drawer page:
 _The VirtualMachine **will have** {{sockets}} sockets, {{threads}} threads, and {{cores}} cores._

Fix confusing "CPU | Memory" helper texts in the following pages (there was incorrect resource in the text):
- _Review and create VirtualMachine_ page:
 _The VirtualMachine **will have** {{sockets}} sockets, {{threads}} threads, and {{cores}} cores._
- VM _Template_ _Details_ tab:
 _The **template** has {{sockets}} sockets, {{threads}} threads, and {{cores}} cores._
- VMI _Details_: 
 _The **VirtualMachineInstance** has {{sockets}} sockets, {{threads}} threads, and {{cores}} cores._

In which pages the help text for "CPU | Memory" field remains the same/what is correct:
- VM _Details_ page

What else has changed (see https://bugzilla.redhat.com/show_bug.cgi?id=2172390#c20):
- _CPUs = sockets X threads X cores._ in the helper text changed to:
 _CPUs = sockets **x** threads **x** cores._

TODO:
- [x] wait for Avital to make sure about the help text for Templates

## 🎥 Screenshots
**Before:**
_Catalog_ drawer - no any help text available:
catalog_before

VM _Template_ _Details_ tab (looks same also in VMI _Details_ tab and in _Review and create VirtualMachine_ page):
template_before

**After:**
_Catalog_ drawer:
drawerr

_Review and create VirtualMachine_ page:
review

VM _Template_ _Details_ tab:
templatee

VMI _Details_: 
vmi

### CODE REVIEW

**Comment by hstastna** [comment]
@apinnick WDYT about these changes? WDYT about the help text for templates?  'The VirtualMachineTemplate has...' or 'The Template has...' or 'The template has...'  is the best? Thanks!

File: src/utils/components/CPUDescription/utils/types.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { CpuMemHelperTextResources } from './utils';
+
+export type CpuMemHelperTextResourcesProps =
+  | typeof CpuMemHelperTextResources.ActualVM
```
Comment (metalice) [inline src/utils/components/CPUDescription/utils/types.ts:4]: isnt all typeof CpuMemHelperTextResources.ActualVM
  | typeof CpuMemHelperTextResources.FutureVM
  | typeof CpuMemHelperTextResources.Template
  | typeof CpuMemHelperTextResources.VMI;

is just type string?

Comment (metalice) [reply src/utils/components/CPUDescription/utils/types.ts:4]: maybe u meant to do type CpuMemHelperTextResourcesProps = keyof CpuMemHelperTextResources ?

Comment (hstastna) [reply src/utils/components/CPUDescription/utils/types.ts:4]: Yeah, that could work, too. Just haven't realized that. Let me try. And thanks!

Comment (hstastna) [reply src/utils/components/CPUDescription/utils/types.ts:4]: Soo, no, it doesn't work.

Comment (metalice) [reply src/utils/components/CPUDescription/utils/types.ts:4]: So I think u can simply use string.. As they are string

Comment (hstastna) [reply src/utils/components/CPUDescription/utils/types.ts:4]: But if I use just `string` as a type, the scope of possible values changes dramatically. This is not what I wanted to achieve.

Comment (metalice) [reply src/utils/components/CPUDescription/utils/types.ts:4]: what is currently will respolve to string and wont server the purpose u trying to achieve 
image

even when I tried something a little bit more complicated such as 
image

it won't work beacust of t function that resolves to string but typescript don't know to which exact string

so maybe try to search for something that will serve your purpose or just use type string.

Comment (hstastna) [reply src/utils/components/CPUDescription/utils/types.ts:4]: ok

**Comment by hstastna** [comment]
@avivtur @metalice @pcbailey @upalatucci @vojtechszocs please review

**Pull request merged**